src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrForwardFlower.cs

```diff
@@ -216,6 +216,7 @@ await _vmrCloneManager.PrepareVmrAsync(
             LastFlows lastFlows = await GetLastFlowsAsync(mapping, sourceRepo, currentIsBackflow: false);
             await vmr.CheckoutAsync(lastFlows.LastFlow.VmrSha);
             await vmr.CreateBranchAsync(headBranch, overwriteExistingBranch: true);
+            await _dependencyTracker.RefreshMetadataAsync();
```

Comment:
oh was this the missing thing? Weird that it didn't cause more problems

Reply:
Yeah, I can think of other cases where this could be super dangerous. I am inclined to rolling this out today even.